### PR TITLE
Adding support for mapping Jira LexoRank

### PR DESCRIPTION
**Pull request by izzy32457**
Ensures backlog ordering is kept when migrating Jira issues to WI.
Also added some warning logging messages when this process could cause invalid ordering due to Jira LexoRank needing to be re-balanced.

Only basic unit tests added.
Not sure if JiraMapper needs some extra tests added?

### CODE REVIEW

File: src/WorkItemMigrator/tests/Migration.Jira-Export.Tests/RevisionUtils/FieldMapperUtilsTests.cs [hunk 1]
```diff
@@ -409,6 +409,20 @@ public void When_calling_map_rendered_value_with_null_arguments_Then_and_excepti
             Assert.Throws<ArgumentNullException>(() => { FieldMapperUtils.MapRenderedValue(null, null, false, null, null); });
         }
 
+        [TestCase("2|hzyxfj:", 1088341183.0)]
+        [TestCase("2|hzyxfj:rx4", 1088341183.36184)]
+        public void When_calling_map_lexorank_value_with_valid_argument_Then_max_value_is_returned(string lexoRank, decimal expectedRank)
```
Comment (moerketh) [inline src/WorkItemMigrator/tests/Migration.Jira-Export.Tests/RevisionUtils/FieldMapperUtilsTests.cs:414]: Should this be "then the correct value is returned"?

File: src/WorkItemMigrator/JiraExport/RevisionUtils/FieldMapperUtils.cs [hunk 2]
```diff
@@ -158,6 +160,43 @@ public static object MapSprint(string iterationPathsString)
...
+            var lexoSplit = lexoRank.Split(new[] {'|', ':'}, StringSplitOptions.RemoveEmptyEntries);
+
+            // calculate the numeric value of the rank and sub-rank (if available)
+            var b36Rank = Base36.Decode(lexoSplit[1]);
+            var b36SubRank = lexoSplit.Length == 3 && !string.IsNullOrEmpty(lexoSplit[2])
+                ? Base36.Decode(lexoSplit[2])
+                : 0L;
+
+            // calculate final rank value
+            var rank = Convert.ToDecimal($"{b36Rank}.{b36SubRank}");
```
Comment (moerketh) [inline src/WorkItemMigrator/JiraExport/RevisionUtils/FieldMapperUtils.cs:190]: This doesn't work for me (the tests fail) unless I change the dot to a comma to match my culture settings. Suggested fix: 
`var rank = Convert.ToDecimal($"{b36Rank}.{b36SubRank}", CultureInfo.InvariantCulture.NumberFormat);`

File: src/WorkItemMigrator/JiraExport/RevisionUtils/FieldMapperUtils.cs [hunk 3]
```diff
@@ -158,6 +160,43 @@ public static object MapSprint(string iterationPathsString)
...
+            RegexOptions.Compiled | RegexOptions.CultureInvariant | RegexOptions.Singleline);
+
+        public static object MapLexoRank(string lexoRank)
+        {
+            if (string.IsNullOrEmpty(lexoRank) || !LexoRankRegex.IsMatch(lexoRank))
+                return decimal.MaxValue;
+
+            if (CalculatedLexoRanks.ContainsKey(lexoRank))
+            {
+                Logger.Log(LogLevel.Warning, "Duplicate rank detected. You may need to re-balance the JIRA LexoRank. see: https://confluence.atlassian.com/adminjiraserver/managing-lexorank-938847803.html");
```
Comment (moerketh) [inline src/WorkItemMigrator/JiraExport/RevisionUtils/FieldMapperUtils.cs:176]: I think exceptions that actually tell me what to do to fix them are fantastic :)

File: docs/config.md [hunk 4]
```diff
@@ -100,6 +100,7 @@ Currently the tool has a rather naive implementation for mapping certain constru
 |MapArray|Maps an array by replacing comma with semi-colon|
 |MapRemainingWork|Maps and converts a Jira time to hours|
 |MapRendered|Maps field to rendered html format value|
+|MapLexoRank|Maps and converts a Jira LexoRank to decimal. When mapping this type of field, ensure the correct Jira custom field is used and mapped to the relevant Azure DevOps prioritization field (see: https://learn.microsoft.com/en-us/azure/devops/boards/queries/planning-ranking-priorities?view=azure-devops#fields-used-to-plan-and-prioritize-work)|
 |(default)|Simply copies soure to target|
 
 ## Example configuration
```
Comment (moerketh) [inline docs/config.md:106]: Perhaps we can do a little note here that this example applies to the Scrum template? Agile and CMMI should use StackRank.

**Review by moerketh — COMMENTED** [review]
Thanks! I'm giving this PR a try. This would make some Product Owners on my side very happy. I added some comments.

File: src/WorkItemMigrator/tests/Migration.Jira-Export.Tests/RevisionUtils/FieldMapperUtilsTests.cs (hunk 1 above)
Comment (izzycoding) [reply src/WorkItemMigrator/tests/Migration.Jira-Export.Tests/RevisionUtils/FieldMapperUtilsTests.cs:414]: Yes it should. Will update that when in the office.

File: src/WorkItemMigrator/JiraExport/RevisionUtils/FieldMapperUtils.cs (hunk 2 above)
Comment (izzycoding) [reply src/WorkItemMigrator/JiraExport/RevisionUtils/FieldMapperUtils.cs:190]: Will make suggested change

File: src/WorkItemMigrator/JiraExport/RevisionUtils/FieldMapperUtils.cs (hunk 3 above)
Comment (izzycoding) [reply src/WorkItemMigrator/JiraExport/RevisionUtils/FieldMapperUtils.cs:176]: Always do this as I kept loosing the link while trying to get this working. Lol

File: docs/config.md (hunk 4 above)
Comment (izzycoding) [reply docs/config.md:106]: Can’t see the rest of the file after this comment, but doesn’t the config specify “scrum” as the process? Maybe add an alternative example that is not scrum to show this?

File: src/WorkItemMigrator/tests/Migration.Jira-Export.Tests/RevisionUtils/FieldMapperUtilsTests.cs (hunk 1 above)
Comment (izzy32457) [reply src/WorkItemMigrator/tests/Migration.Jira-Export.Tests/RevisionUtils/FieldMapperUtilsTests.cs]: This should be updated now

File: src/WorkItemMigrator/JiraExport/RevisionUtils/FieldMapperUtils.cs (hunk 2 above)
Comment (izzy32457) [reply src/WorkItemMigrator/JiraExport/RevisionUtils/FieldMapperUtils.cs]: This should be updated now

**Pull request merged**